web/src/layouts/MainLayout/MainLayout.js

```diff
@@ -14,23 +16,35 @@ const MainLayout = ({ children }) => {
     Passing the authenticated user to StripeProvider for use for Stripe Checkout and Stripe Portals
     User needs to manage authentication of customers
   */
-  const { getCurrentUser } = useAuth()
+  const { getCurrentUser, isAuthenticated } = useAuth()
   const { addStripeId } = useStripeId()
+  const [customer, setCustomer] = useState('')
+
+  useEffect(() => {
+    const getUserStripeId = async () => {
+      const { id, stripeId } = await getCurrentUser()
+      if (!stripeId) {
+        const user = await addStripeId(id)
```

Comment:
Use a addUserStripeId hook to find the Stripe Id of a user using their email on the backend.